src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/ImpISQLServerBulkRecord_IssuesTest.java

```diff
@@ -0,0 +1,435 @@
+
+    static Statement stmt = null;
+    static PreparedStatement pStmt = null;
+    static String query;
+    static SQLServerConnection con = null;
+    static String srcTable = "sourceTable";
+    static String destTable = "destTable";
+    String variation;
+
+    /**
+     * Testing that sending a bigger varchar(3) to varchar(2) is thowing the proper error message.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testVarchar() throws Exception {
+        variation = "testVarchar";
+        BulkDat bData = new BulkDat(variation);
+        String value = "aa";
+        query = "CREATE TABLE " + destTable + " (smallDATA varchar(2))";
+        stmt.executeUpdate(query);
+
+        try {
+            SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+            bcOperation.setDestinationTableName(destTable);
+            bcOperation.writeToServer(bData);
+            bcOperation.close();
+        }
+        catch (Exception e) {
+            if (e instanceof SQLServerException) {
+                assertTrue(e.getMessage().contains("The given value of type"), "Invalid Error message: " + e.toString());
+            }
+        }
+        ResultSet rs = stmt.executeQuery("select * from " + destTable);
+        while (rs.next()) {
+            assertEquals(rs.getString(1), value);
+        }
+    }
+
+    /**
+     * Testing that setting scale and precision 0 in column meta data for smalldatetime should work
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testSmalldatetime() throws Exception {
+        variation = "testSmalldatetime";
+        BulkDat bData = new BulkDat(variation);
+        String value = ("1954-05-22 02:44:00.0").toString();
+        query = "CREATE TABLE " + destTable + " (smallDATA smalldatetime)";
+        stmt.executeUpdate(query);
+
+        SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+        bcOperation.setDestinationTableName(destTable);
+        bcOperation.writeToServer(bData);
+        bcOperation.close();
+
+        ResultSet rs = stmt.executeQuery("select * from " + destTable);
+        while (rs.next()) {
+            assertEquals(rs.getString(1), value);
+        }
+    }
+
+    /**
+     * Testing that setting out of range value for small datetime is throwing the proper message
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testSmalldatetimeOutofRange() throws Exception {
+        variation = "testSmalldatetimeOutofRange";
+        BulkDat bData = new BulkDat(variation);
+        String value = ("1954-05-22 02:44:00.0").toString();
+
+        query = "CREATE TABLE " + destTable + " (smallDATA smalldatetime)";
+        stmt.executeUpdate(query);
+
+        try {
+            SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+            bcOperation.setDestinationTableName(destTable);
+            bcOperation.writeToServer(bData);
+            bcOperation.close();
```

Comment:
after bcOperation.close(), you should force the test to fail with something like:
`fail("BulkCopy executed when it it was expected to fail");`

Once this is done, you can remove the section after the catch that iterates through the resultset